File: in_Java/1408 String Matching in an Array.java
class Solution {
    public List<String> stringMatching(String[] words) {
        List<String> res = new ArrayList<>();
        for(int i = 0; i < words.length; i++){
            String cur = words[i];
            for(int j = 0; j < words.length; j++){
                if(j == i) continue;
                String now = words[j];
                if(now.length() < cur.length()) continue;
                if(check(cur,now)){
                    res.add(cur);
                    break;
                }
            }
        }
        return res;
    }
    private boolean check(String a, String b){
        int n = a.length();
        for(int i = 0; i <= b.length() - n; i++){
            if(b.substring(i,i+n).equals(a)) return true;
        }
        return false;
    }
}

File: in_Python/1408 String Matching in an Array.py
class Solution(object):
    def stringMatching(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        words.sort(key = len)
        res = []
        for i in range(len(words)):
            for j in range(i+1, len(words)):
                if words[i] in words[j]:
                    if words[i] not in res:
                        res.append(words[i])
        return res